music-journal/pages/api/recent-tracks-from-spotify.js

```diff
@@ -1,85 +1,45 @@
-const firebaseAdmin = require('firebase-admin');
-const firebase = require('firebase');
-const serviceAccount = require('../../service-account.json');
-const SpotifyWebApi = require('spotify-web-api-node');
+import { auth } from 'firebase';
 import Cors from 'cors';
-const cors = Cors({
-    methods: ['GET', 'HEAD'],
-});
+import { database } from '../../utils/initFirebaseAdmin';
+import initMiddleware from '../../utils/initMiddleware'
+import Spotify from '../../utils/initSpotify';
 
-// Helper method to wait for a middleware to execute before continuing
-// And to throw an error when an error happens in a middleware
-function runMiddleware(req, res, fn) {
-    return new Promise((resolve, reject) => {
-        fn(req, res, (result) => {
-            if (result instanceof Error) {
-                return reject(result)
-            }
-
-            return resolve(result)
-        })
+const cors = initMiddleware(
+    Cors({
+        methods: ['GET', 'POST', 'OPTIONS', 'HEAD'],
     })
-}
+)
 
+async function handler(req, res) {
+    await cors(req, res)
 
-if (!firebaseAdmin.apps.length) {
-    firebaseAdmin.initializeApp({
-        credential: firebaseAdmin.credential.cert(serviceAccount),
-        databaseURL: process.env.FIREBASE_DATABASE_URL,
-    });
-}
-
-const Spotify = new SpotifyWebApi({
-    clientId: process.env.SPOTIFY_CLIENT_ID,
-    clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
-    redirectUri: process.env.AUTH_REDIRECT_URL,
-});
-
-export default async function handler(req, res) {
-    await runMiddleware(req, res, cors);
     if (!req.query.token) {
         return res.writeHead(401).end();
     }
-
-    let loginFailed = false;
+    
     try {
-        await firebase.auth().signInWithCustomToken(req.query.token);
+        const user = auth().currentUser;
```

Comment:
I see what you mean, I think it's right to do it this way as otherwise you could end up with residual auth state on the server cross-populating between clients. Maybe this could return a 503 if there isn't a current user? I think it will work as is but will throw error logs on the server if this endpoint is somehow accessed by an unauthenticated user

Reply:
Looks like the verify token middleware will actual deal with it properly so we can throw the error there